### PR TITLE
Initial support for nRF22xx devices

### PR DESCRIPTION
**Pull request by ankuns**
This PR adds initial support for the nRF22xx devices for SoCs families nRF52 and nRF53.
MPSL libraries support provided in PRs:
https://github.com/nrfconnect/sdk-nrfxlib/pull/1328
https://github.com/nrfconnect/sdk-nrf/pull/15217

Intentionally no doc yet.

### CODE REVIEW

File: boards/shields/nrf2220ek/nrf2220ek.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CONFIG_BOOT_BANNER=n
```
Comment (jciupis) [inline boards/shields/nrf2220ek/nrf2220ek.conf:1]: Are these options necessary for nRF52 series too?

File: boards/shields/nrf2220ek/nrf2220ek_fwd.overlay [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * For instance, consider an application targeted for nRF5340 application core that specifies
+ * a child image targeted for nRF5340 network core, which drives the nRF2220 Front-End Module.
+ * The shield overlays could be provided to the build using the following command:
+ *
+ * west build -p -b nrf5340dk/nrf5340/cpuapp -- -DSHIELD=nrf2220ek_fwd -D<child_image>_SHIELD=nrf2220ek
+ */
+
+&gpio_fwd {
+	nrf2220-gpio-if {
+		gpios = <&arduino_header 14 0>,		/* cs-gpios */
```
Comment (jciupis) [inline boards/shields/nrf2220ek/nrf2220ek_fwd.overlay:19]: Please align comments

File: boards/shields/nrf2240ek/nrf2240ek.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CONFIG_BOOT_BANNER=n
```
Comment (jciupis) [inline boards/shields/nrf2240ek/nrf2240ek.conf:1]: Same question as for nRF2220: are these options necessary for nRF52 series too?

File: boards/shields/nrf2240ek/nrf2240ek.overlay [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* Pin mapping suitable for use nRF2240-EK PCA63564 with
+ * Nordic Interposer Board A PCA64172 v0.2.0.
+ * The nRF2240-EK in SLOT 2 of PCA64172.
```
Comment (jciupis) [inline boards/shields/nrf2240ek/nrf2240ek.overlay:8]:
```suggestion
 * Nordic Interposer Board A PCA63564 v0.2.0.
 * The nRF2240-EK in SLOT 2 of PCA63564.
```

File: boards/shields/nrf2240ek/nrf2240ek_fwd.overlay [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * For instance, consider an application targeted for nRF5340 application core that specifies
+ * a child image targeted for nRF5340 network core, which drives the nRF2240 Front-End Module.
+ * The shield overlays could be provided to the build using the following command:
+ *
+ * west build -p -b nrf5340dk/nrf5340/cpuapp -- -DSHIELD=nrf2240ek_fwd -D<child_image>_SHIELD=nrf2240ek
+ */
+
+&gpio_fwd {
+	nrf2240-gpio-if {
+		gpios = <&arduino_header 14 0>,		/* cs-gpios */
```
Comment (jciupis) [inline boards/shields/nrf2240ek/nrf2240ek_fwd.overlay:19]: Please align comments

File: subsys/mpsl/fem/nrf2240/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_library_sources(mpsl_fem_nrf2240.c)
```
Comment (jciupis) [inline subsys/mpsl/fem/nrf2240/CMakeLists.txt:7]: Isn't `nrfx_twim_stub.c` needed here as well, exactly as for nRF2220?

File: subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+}
+
+static int mpsl_fem_init(void)
+{
+	mpsl_fem_device_config_254_apply_set(IS_ENABLED(CONFIG_MPSL_FEM_DEVICE_CONFIG_254));
+
+	return fem_nrf2240_configure();
+}
+
+SYS_INIT(mpsl_fem_init, POST_KERNEL, 49);
```
Comment (jciupis) [inline subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c:249]: Applies both here and for nRF2240: why this particular priority? It looks very specific

File: boards/shields/nrf2240ek/nrf2240ek.overlay (hunk 4 above)
Comment (ankuns) [reply boards/shields/nrf2240ek/nrf2240ek.overlay:8]: Your proposal is incorrect. The Nordic Interposer Board A is PCA64172. The nRF2240-EK goes into SLOT2 of the Interposer Board so PCA64172 here is correct. The PCA63564 is the nRF2240-EK itself not the interposer board.

Comment (jciupis) [reply boards/shields/nrf2240ek/nrf2240ek.overlay:8]: Oops, you're right!

File: boards/shields/nrf2220ek/nrf2220ek.conf (hunk 1 above)
Comment (jciupis) [reply boards/shields/nrf2220ek/nrf2220ek.conf:1]: This configuration is needed unconditionally for nRF53 and in some cases for nRF52 too. After F2F discussion we decided to keep it as a limitation.

File: boards/shields/nrf2240ek/nrf2240ek.conf (hunk 3 above)
Comment (jciupis) [reply boards/shields/nrf2240ek/nrf2240ek.conf:1]: This configuration is needed unconditionally for nRF53 and in some cases for nRF52 too. After F2F discussion we decided to keep it as a limitation.

File: boards/shields/nrf2220ek/nrf2220ek_fwd.overlay (hunk 2 above)
Comment (ankuns) [reply boards/shields/nrf2220ek/nrf2220ek_fwd.overlay]: done

File: boards/shields/nrf2240ek/nrf2240ek_fwd.overlay (hunk 5 above)
Comment (ankuns) [reply boards/shields/nrf2240ek/nrf2240ek_fwd.overlay]: done

File: subsys/mpsl/fem/nrf2240/CMakeLists.txt (hunk 6 above)
Comment (ankuns) [reply subsys/mpsl/fem/nrf2240/CMakeLists.txt:7]: Please have a look on introduced now Kconfig `MPSL_FEM_USE_TWIM_STUB`. This option turns on the build of `nrfx_twim_stub.c` which is moved to `subsys/mpsl/fem/common/`

File: subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c (hunk 7 above)
Comment (ankuns) [reply subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c:251]: Leftover related to possibility to use CONFIG_I2C_SHELL, I will come with solution.

**Review by jciupis — APPROVED** [review]

File: subsys/mpsl/fem/nrf2240/CMakeLists.txt (hunk 6 above)
Comment (jciupis) [reply subsys/mpsl/fem/nrf2240/CMakeLists.txt:7]: Perfect

File: subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c (hunk 7 above)
Comment (ankuns) [reply subsys/mpsl/fem/nrf2240/mpsl_fem_nrf2240.c]: Solved in e68e23394d0f272f5d54059d051dc14fd4903d7a

**Comment by jciupis** [comment]
All good 👍

**Review by piotrkoziar — APPROVED** [review]

**Pull request merged**